### PR TITLE
[Storage] Add get_state_value API for supporting tables in StateView

### PR DESCRIPTION
**Pull request by sitalkedia**
## Motivation

Adding support for getting state value from state view - this will be used for table support as well as fine grained storage going forward. There are some TODOs in the diff that needs to be addressed when we make changes to the writeset for tables. 

## Test Plan

Existing UTs

### CODE REVIEW

File: storage/storage-interface/src/state_view.rs [hunk 1]
```diff
@@ -182,18 +165,59 @@ impl StateView for VerifiedStateView {
...
+        // Update the cache if still empty
+        let mut cache = self.state_key_value_cache.write();
+        let new_value = cache
+            .entry(state_key.clone())
+            .or_insert_with(|| state_store_value_option.unwrap_or_default());
+
+        Ok(Some(new_value.bytes.clone()))
+    }
+
+    fn get_account_resource(&self, access_path: &AccessPath) -> Result<Option<Vec<u8>>> {
```
Comment (msmouse) [inline storage/storage-interface/src/state_view.rs:198]: This works for modules as well, so the name isn't correct?

File: aptos-move/aptos-vm/src/data_cache.rs [hunk 2]
```diff
@@ -98,6 +99,27 @@ impl<'block, S: StateView> StateView for StateViewCache<'block, S> {
         }
     }
 
+    fn get_state_value(&self, state_key: &StateKey) -> anyhow::Result<Option<Vec<u8>>> {
+        //TODO: Add a caching layer on this once the VM write set starts populating state_value changes.
```
Comment (msmouse) [inline aptos-move/aptos-vm/src/data_cache.rs:103]: yeah.. we need it soon-ish

File: storage/storage-interface/src/state_view.rs (hunk 1 above)
Comment (sitalkedia) [reply storage/storage-interface/src/state_view.rs:198]: Good point, any suggestion on name?

Comment (msmouse) [reply storage/storage-interface/src/state_view.rs:198]: `get_by_access_path()` :)

File: aptos-move/aptos-vm/src/data_cache.rs (hunk 2 above)
Comment (sitalkedia) [reply aptos-move/aptos-vm/src/data_cache.rs:103]: Once we have write set API changes for table, we can add this.

**Review by msmouse — APPROVED** [review]

File: aptos-move/aptos-vm/src/data_cache.rs [hunk 3]
```diff
@@ -98,6 +99,27 @@ impl<'block, S: StateView> StateView for StateViewCache<'block, S> {
         }
     }
 
+    fn get_state_value(&self, state_key: &StateKey) -> anyhow::Result<Option<Vec<u8>>> {
```
Comment (msmouse) [inline aptos-move/aptos-vm/src/data_cache.rs:102]: I slightly prefer to call this just `get()`, just saying :P

Comment (sitalkedia) [reply aptos-move/aptos-vm/src/data_cache.rs:102]: I would like that too but at this point that would be ambigous. Let's rename this to `get()` once we completely move to the fine-grained storage :)

**Pull request closed without merging**